### PR TITLE
Ports Banjo Kazooie

### PR DESCRIPTION
**Pull request by Altoids1**
https://youtu.be/R_mkkK-L1qU?t=64

I recently dug up this ``chatter`` proc, abandoned and technically unused. I have since converted it into an adminbus command, and intend on using it to implement audio cues for important speech (spellcasting by witches, speech coming from a baron or even a knight, etc etc). Hopefully this can be one of those little things that help make our server feel unique.

Here's a checklist for all that:
- [x] Adminbus command
- [ ] Witch Honking
- [ ] Some pre-written stuff to handle having job-specific chattering (such as for the Baron)

### CODE REVIEW

**Comment by morrowwolf** [comment]
I'm gonna just preemptively put this on the next meeting's docket.

**Comment by Parune** [comment]
Is this like an alternative to text to speech?  So when someone talks, they make the squaking/chattering sound?

**Comment by morrowwolf** [comment]
From what I understand yes.

**Comment by Altoids1** [comment]
I didn't want to paste the whole of the proc I found on here, since it's a bit long and I've done it before in #dev-code , but y'all don't seem to know what this does, so...

**TL;DR:** Chatter takes in a message, finds the length of each word (as well as any punctuation that might indicate a pause), and plays a series of sounds, with the length of each sound correlating to the length of the words, as a sort of simplified text-to-speech.
```dm
/proc/chatter(message, phomeme, atom/A)
	// We want to transform any message into a list of numbers
	// and punctuation marks
	// For example:
	// "Hi." -> [2, '.']
	// "HALP GEROGE MELLONS, that swine, is GRIFFIN ME!"
	// -> [4, 6, 7, ',', 4, 5, ',', '2', 7, 2, '!']
	// "fuck,thissentenceissquashed" -> [4, ',', 21]

	var/list/punctuation = list(",",":",";",".","?","!","\'","-")
	var/regex/R = regex("(\[\\l\\d]*)(\[^\\l\\d\\s])?", "g")
	var/list/letter_count = list()
	while(R.Find(message) != 0)
		if(R.group[1])
			letter_count += length(R.group[1])
		if(R.group[2])
			letter_count += R.group[2]

	spawn(0)
		for(var/item in letter_count)
			if (item in punctuation)
				// simulate pausing in talking
				// ignore semi-colons because of their use in HTML escaping
				if (item in list(",", ":"))
					sleep(3)
				if (item in list("!", "?", "."))
					sleep(6)
				continue

			if(isnum(item))
				var/length = min(item, 10)
				if (length == 0)
					// "verbalise" long spaces
					sleep(1)
				chatter_speak_word(A.loc, phomeme, length)
```
```dm
/proc/chatter_speak_word(loc, phomeme, length)
	var/path = "sound/chatter/[phomeme]_[length].ogg"

	playsound(loc, path,
		vol = 40, vary = 0, extrarange = 3, falloff = FALSE)

	sleep((length + 1) * chatter_get_sleep_multiplier(phomeme))

/proc/chatter_get_sleep_multiplier(phomeme)
	// These values are tenths of seconds, so 0.5 == 0.05seconds
	. = 1
	switch(phomeme)
		if("papyrus")
			. = 0.5
		if("griffin")
			. = 0.5
		if("sans")
			. = 0.7
		if("owl")
			. = 0.7
```

The actual sounds that play depend on what's in the ``/sound/chatter`` folder. The sounds in that folder right now are for the "griffin" and "owl" voices, and they only support word lengths between 1 and 10; larger words just play the 10th sound.
